components/patient-data/rest/src/test/java/org/phenotips/data/rest/internal/DefaultPatientsResourceImplTest.java

```diff
@@ -0,0 +1,224 @@
+        doReturn(true).when(this.access).hasAccess(eq(Right.EDIT), any(DocumentReference.class), any(EntityReference.class));
+        doReturn(this.patient).when(this.repository).createNewPatient();
+        JSONObject jsonPatient = new JSONObject();
+        Response response = this.patientsResource.addPatient(jsonPatient.toString());
+        Assert.assertEquals(Response.Status.CREATED.getStatusCode(), response.getStatus());
+        verify(this.logger).debug("Importing new patient from JSON via REST: {}", jsonPatient.toString());
+    }
+
+    @Test
+    public void listPatientsNullOrderField() throws QueryException {
+        Query query = mock(DefaultQuery.class);
+        doReturn(query).when(this.queries).createQuery(anyString(), anyString());
+        doReturn(query).when(query).bindValue(anyString(), anyString());
+        doReturn(new ArrayList<Object[]>()).when(query).execute();
+        Patients result = this.patientsResource.listPatients(0, 30, null, "asc");
+        verify(this.queries).createQuery("select doc.fullName, p.external_id, doc.creator, doc.creationDate, doc.version, doc.author, doc.date"
+                + " from Document doc, doc.object(PhenoTips.PatientClass) p where doc.name <> :t order by "
+                + "doc.name" + " asc", "xwql");
+    }
+
+    @Test
+    public void listPatientsNullOrder() throws QueryException {
+        Query query = mock(DefaultQuery.class);
+        doReturn(query).when(this.queries).createQuery(anyString(), anyString());
+        doReturn(query).when(query).bindValue(anyString(), anyString());
+        doReturn(new ArrayList<Object[]>()).when(query).execute();
+        Patients result = this.patientsResource.listPatients(0, 30, "id", null);
+        verify(this.queries).createQuery("select doc.fullName, p.external_id, doc.creator, doc.creationDate, doc.version, doc.author, doc.date"
+                + " from Document doc, doc.object(PhenoTips.PatientClass) p where doc.name <> :t order by "
+                + "doc.name" + " asc", "xwql");
+    }
+
+    @Test
+    public void listPatientsNonDefaultBehaviour() throws WebApplicationException, QueryException {
+        Query query = mock(DefaultQuery.class);
+        doReturn(query).when(this.queries).createQuery(anyString(), anyString());
+        doReturn(query).when(query).bindValue(anyString(), anyString());
+        doReturn(new ArrayList<Object[]>()).when(query).execute();
+        Patients result = this.patientsResource.listPatients(0, 30, "eid", "desc");
+        verify(this.queries).createQuery("select doc.fullName, p.external_id, doc.creator, doc.creationDate, doc.version, doc.author, doc.date"
+                + " from Document doc, doc.object(PhenoTips.PatientClass) p where doc.name <> :t order by "
+                + "p.external_id" + " desc", "xwql");
+    }
+
+    @Test
+    public void listPatientFailureHandling() throws QueryException {
+        WebApplicationException exception = new WebApplicationException();
```

Comment:
I don't think creating a new `WebApplicationException` object is a good idea. Right now your tests are dependant on the implementation of the default constructor of `WebApplicationException`. One way that this could be problematic is if the status code is, by default, set to the same value you are testing for. Then you wouldn't be able to tell if it was this constructor, or the method you are testing that set that status code.
That's exactly what happens in this test. the `WebApplicationException` status code is set to `INTERNAL_SERVER_ERROR`, so this test will pass, whether or not the `WebApplicationException` is actually thrown.
Additionally, in the event that `listPatients()` doesn't throw a `WebApplicationException` under the condition that you have set up, this test will effectively give feedback that there was a `WebApplicationException` but it used the wrong status code. Which could be quite confusing.
To solve this problem you could either use the `@Test(expected = WebApplicationException.class)` notation and check the logger behaviour in another test, or use try/catch as you did here but initialize this value to `null`, set it to the caught exception and check whether it is still `null` afterward. This is the solution I used [here](https://github.com/phenotips/phenotips/pull/1802/files#diff-d4870c253487d0d0c680ad7578d14616R227)